rest/OrderService.js

```diff
@@ -2,35 +2,66 @@ import ProductPersistence from "../persistence/ProductPersistence.js";
 import OrderPersistence from "../persistence/OrderPersistence.js";
 import IllegalArgumentError from "./IllegalArgumentError.js";
 
+const EMPTY_ARGUMENT_ERR = "Body contains an empty argument!";
+const INVALID_TIMESTAMP_ERR = "Timestamp create should be now!";
+
 class OrderService {
-    #productPersistence;
     #orderPersistence;
 
     constructor() {
-        this.#productPersistence = new ProductPersistence();
         this.#orderPersistence = new OrderPersistence();
     }
 
     async createOrder(order) {
-        return await this.#orderPersistence.create(order);
+        this.#assertNoEmptyArgs(order);
+        this.#assertValidTimestamp(order.timestampCreated);
+        return await this.#orderPersistence.create(
+            this.#buildOrderObject(order));
     }
 
     async getOrderById(orderId) {
         return await this.#orderPersistence.getById(orderId);
     }
 
+    async getOrderByProductId(productId) {
```

Comment:
Shouldn't this be getOrders?

Assuming you use findAll in the persistence class it should return an array.